src/libraries/System.CodeDom/tests/System/CodeDom/Compiler/CSharpCodeGeneratorTests.cs

```diff
@@ -0,0 +1,2728 @@
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(IsValidIdentifier_TestData))]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, ".NET Framework is missing some newer keywords")]
+        public void IsValidIdentifier_Invoke_ReturnsExpected(string value, bool expected)
+        {
+            ICodeGenerator generator = GetGenerator();
+            Assert.Equal(expected, generator.IsValidIdentifier(value));
+        }
+
+        public static IEnumerable<object[]> GenerateCodeFromExpression_TestData()
+        {
+            string nl = Environment.NewLine;
+            var customOptions = new CodeGeneratorOptions
+            {
+                IndentString = "$",
+                ElseOnClosing = true,
+                BracingStyle = "C"
+            };
+
+            // CodeArrayCreateExpression.
+            yield return new object[] { new CodeArrayCreateExpression(), null, "new void[0]" };
+            yield return new object[] { new CodeArrayCreateExpression(new CodeTypeReference("type")), null, "new type[0]" };
+            yield return new object[] { new CodeArrayCreateExpression(new CodeTypeReference("type", 2)), null, "new type[0]" };
+            yield return new object[] { new CodeArrayCreateExpression(new CodeTypeReference(new CodeTypeReference("type", 1), 1)), null, "new type[0][]" };
+
+            yield return new object[] { new CodeArrayCreateExpression(new CodeTypeReference("type"), new CodeExpression[] { new CodePrimitiveExpression(1) }), null, $"new type[] {{{nl}        1}}" };
+            yield return new object[] { new CodeArrayCreateExpression(new CodeTypeReference("type"), new CodeExpression[] { new CodePrimitiveExpression(1), new CodePrimitiveExpression(2) }), null, $"new type[] {{{nl}        1,{nl}        2}}" };
+            yield return new object[] { new CodeArrayCreateExpression(new CodeTypeReference("type"), new CodeExpression[] { new CodePrimitiveExpression(1), new CodePrimitiveExpression(2) }) { SizeExpression = new CodeExpression() }, null, $"new type[] {{{nl}        1,{nl}        2}}" };
+            yield return new object[] { new CodeArrayCreateExpression(new CodeTypeReference("type", 2), new CodeExpression[] { new CodePrimitiveExpression(1) }), null, $"new type[,] {{{nl}        1}}" };
+            yield return new object[] { new CodeArrayCreateExpression(new CodeTypeReference(new CodeTypeReference("type", 1), 1), new CodeExpression[] { new CodePrimitiveExpression(1) }), null, $"new type[][] {{{nl}        1}}" };
+
+            yield return new object[] { new CodeArrayCreateExpression(new CodeTypeReference("type")) { SizeExpression = new CodePrimitiveExpression(1) }, null, "new type[1]" };
+            yield return new object[] { new CodeArrayCreateExpression(new CodeTypeReference("type")) { Size = 0 }, null, "new type[0]" };
+            yield return new object[] { new CodeArrayCreateExpression(new CodeTypeReference("type")) { Size = -1 }, null, "new type[-1]" };
```

Comment:
`new type[-1]` this is odd

Reply:
i see several other expected strings looking odd, so i guess it is by design

Reply:
Yeah i was just adding checks to see if we can generate invalid sizes, which apparently we can as there are no checks!